src/main/java/com/limechain/sync/warpsync/SyncedState.java

```diff
@@ -144,64 +164,11 @@ private void updateState(CommitMessage commitMessage) {
         persistState();
     }
 
-    public void persistState() {
-        List<Pair<String, BigInteger>> authorities = Arrays
-                .stream(authoritySet)
-                .map(authority -> {
-                    String key = authority.getPublicKey().toString();
-                    BigInteger weight = authority.getWeight();
-                    return new Pair<>(key, weight);
-                }).toList();
-
-        StateDto stateDto = new StateDto(
-                latestRound,
-                lastFinalizedBlockHash.toString(),
-                lastFinalizedBlockNumber,
-                authorities,
-                setId
-        );
-        repository.save(DBConstants.SYNC_STATE_KEY, stateDto);
-    }
-
-    public boolean loadState() {
-        Optional<Object> syncState = repository.find(DBConstants.SYNC_STATE_KEY);
-        if (syncState.isPresent() && syncState.get() instanceof StateDto state) {
-            this.latestRound = state.latestRound();
-            this.lastFinalizedBlockHash = Hash256.from(state.lastFinalizedBlockHash());
-            this.lastFinalizedBlockNumber = state.lastFinalizedBlockNumber();
-            this.authoritySet = state.authoritySet()
-                    .stream()
-                    .map(pair -> {
-                        Hash256 publicKey = Hash256.from(pair.getValue0());
-                        BigInteger weight = pair.getValue1();
-                        return new Authority(publicKey, weight);
-                    }).toArray(Authority[]::new);
-
-            this.setId = state.setId();
-
-            return true;
-        }
-        return false;
-    }
-
-    public void saveProofState(byte[][] proof) {
-        repository.save(DBConstants.STATE_TRIE_MERKLE_PROOF, proof);
-        repository.save(DBConstants.STATE_TRIE_ROOT_HASH, stateRoot.toString());
-    }
-
-    public byte[][] loadProof() {
-        return (byte[][]) repository.find(DBConstants.STATE_TRIE_MERKLE_PROOF).orElse(null);
-    }
-
-    public Hash256 loadStateRoot() {
-        Object storedRootState = repository.find(DBConstants.STATE_TRIE_ROOT_HASH).orElse(null);
-        return storedRootState == null ? null : Hash256.from(storedRootState.toString());
-    }
-
     /**
      * Sync our state according to a neighbour message we receive.
      * Try to update our set data (id and authorities) if neighbour has a greater set id than us.
      * Synchronized to avoid race condition between checking and updating set id
```

Comment:
```suggestion
     * Updates the Host's state with information from a neighbour message.
     * Tries to update Host's set data (id and authorities) if neighbour has a greater set id than the Host.
     * Synchronized to avoid race condition between checking and updating set id
```